pagination.go

```diff
@@ -102,3 +124,60 @@ func (c *Client) listHelper(ctx context.Context, pager PagedResponse, opts *List
 	opts.Pages = pages
 	return nil
 }
+
+// flattenQueryStruct flattens a structure into a Resty-compatible query param map.
+// Fields are mapped using the `query` struct tag.
+func flattenQueryStruct(val any) (map[string]string, error) {
+	result := make(map[string]string)
+
+	reflectVal := reflect.ValueOf(val)
+
+	// Deref pointer if necessary
+	if reflectVal.Kind() == reflect.Ptr {
```

Comment:
```suggestion
	if reflectVal.Kind() == reflect.Pointer {
```
[Ptr is the old name for the Pointer kind. ](https://pkg.go.dev/reflect#pkg-constants)